plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/TransactionLogAccess.java

```diff
@@ -269,8 +288,14 @@ private List<AddFileEntry> loadActiveFiles(TableSnapshot tableSnapshot, Metadata
                 typeManager,
                 fileSystemFactory.create(session),
                 fileFormatDataSourceStats,
-                Optional.of(new MetadataAndProtocolEntry(metadataEntry, protocolEntry)))) {
-            return activeAddEntries(checkpointEntries, transactions)
+                Optional.of(new MetadataAndProtocolEntry(metadataEntry, protocolEntry)),
+                partitionConstraint)) {
+            Stream<AddFileEntry> addFileEntryStream = activeAddEntries(checkpointEntries, transactions);
+            if (partitionConstraint.isAll()) {
+                return addFileEntryStream.collect(toImmutableList());
+            }
+            return addFileEntryStream
+                    .filter(addAction -> partitionMatchesPredicate(addAction.getCanonicalPartitionValues(), partitionConstraint.getDomains().orElseThrow()))
```

Comment:
The callers (eg split source) will likely repeat this work, so it's partially wasted.
Still useful because this allows us to materialize a shorter list.

I think this wouldn't be needed here if we could return a Stream/Iterator instead of a List.